### DISCUSSION TITLE
Small problem with the "search" experience

### DISCUSSION THREAD

**Discussion by mreferre** [chipzoller/hugo-clarity#410] [General]
Hi there. I am migrating [my blog](https://it20.info/) to the hugo-clarity theme (I really love it). Everything works just fine except a tiny little thing that I can't seem to be getting right (search). 

Because I have a full working site already I am mostly following [Option #2](https://github.com/chipzoller/hugo-clarity#option-2-hugo-modules) of the instructions. All works fine but this is what I am getting for the search field (note how the search doesn't follow the whole theme): 

![image](https://user-images.githubusercontent.com/4123965/210183780-c6687746-183f-4d87-ac92-77511eea293f.png)

Also, if i try to search for something, instead of results showing up in separate ordered lines, I get a single string that concatenates all the blogs that satisfy my search: 

![image](https://user-images.githubusercontent.com/4123965/210183811-587f7ccb-e7f1-49a4-82b9-e839702260b0.png)

[ These are 5 - 6 separate blogs that satisfy the search term `aws` ] 

I tried to copy the `search.md` file into my `content` folder but that did not change much. 

I am pretty sure this is a user error but I can't quite figure out what's missing in my configuration. Any hint?   

Thanks!

**Comment by chipzoller** [comment]
@onweru we weren't seeing this before, were we?

@mreferre is your theme language set to a non-English language by chance?

**Comment by mreferre** [comment]
Thanks @chipzoller for the fast reply. Nope I saw the comment on search having problems with multi language and that should not be the case here as I have removed entirely the second language for my blog/theme. 

It must be something I’m not doing properly because if I clone the repo, move to the sample site folder and run hugo serve from there with the proper “theme folder” switch per the docs everything works fine (including the search). It’s when I try to apply the theme its sample config to my existing blog that is causing this problem. But I can’t figure out what’s the missing piece I’m leaving behind that is causing the search to behave like that. It almost feels like a CSS thing?

**Comment by onweru** [comment]
Hi @mreferre, can you share a link for your repo where you've applied the clarity theme on your website?

I would be happy to check and advise.

**Comment by mreferre** [comment]
Hi @onweru. Thanks for the folllow up. I am trying to figure out the best way to post my web site on an alternative endpoint but in the meanwhile I have played with it a bit more and I think I found the root cause. For background, I based my migration on the suggestion you had in [Option #2](https://github.com/chipzoller/hugo-clarity#option-2-hugo-modules) and specifically: 

*"If you do already have a site and don't want to risk overwriting anything, we suggest copying the contents of [config](https://github.com/chipzoller/hugo-clarity/blob/master/exampleSite/config) over, as well as replacing your archetypes/post.md (if it exists) with [Hugo Clarity's](https://github.com/chipzoller/hugo-clarity/blob/master/exampleSite/archetypes/post.md). Then migrate any necessary settings from <sitename>/config.toml to <sitename>/config/_default/config.toml and remove the original <sitename>/config.toml file."*

However the example site in your repo has more stuff in it that seem to be required. For example your `resources` folder has a number of `sass` assets that I am missing in my original site. And they seem to the the culprit for the misbehavior I am noting. You should be able to replicate my problem easily by deleting the `sass` folder entirely under the `resources` -> `_gen` -> `assets` folder. 

I can't tell if that folder (`assets`) is part of the theme or the blog as a whole. Being it at the same level of the `themes` folder I would have thought that that was not a dependency for a theme but apparently it is? Sorry I am kind of new to Hugo (well not really super new but I have probably always used it very mechanically without spending much time to learn it) and I am finding hard to draw a line between the general site configuration and the theme configurations. I really liked the notion of themes completely independent and self-contained in their own folder.  

Am I making any sense? Do your instructions need to be expanded to include more than the theme folder + the config folder + archetypes folder? 

***Important caveat***: I did not create a hugo module for the theme and I have rather cloned into the `themes` folder your theme. And then I have configured `<sitename>/config/_default/config.toml` with `theme = "hugo-clarity"` (i.e. I did not configured the GH repo as the `theme` value). I do not see how that would have made a difference for my problem but I figured I'd mention it.

**Reply by onweru** [reply]
We include the resources folder because it's a [requirement](https://github.com/gohugoio/hugoThemes#adding-a-theme-to-the-list) for the theme to be listed.

> I really liked the notion of themes completely independent and self-contained in their own folder.

Absolutely, it makes sense.

I admit, that we made things sound a little too easy under `option 2`. In practice however, things should realistically need making a few edits here and there to map your previous setup to clarity's. And to customize it to your liking of course.

Also a descent amount of comfort with Hugo's [content management](https://gohugo.io/content-management/) approach is assumed.

>  I do not see how that would have made a difference for my problem but I figured I'd mention it.

You're right.

I think if you got all the other clarity's features to work but search, you shouldn't have to make too many changes. You will overturn that obstacle.

**Comment by mreferre** [comment]
I guess my main question right now is (and my confusion is mostly due to my poor understanding of how Hugo core and the various themes interact)... am I supposed to fill my core folder structure with the files shipped with the theme? For example, in addition to the files in the `resources` folder (that seem to be the culprit for breaking the search experience) there are a number of other folders in the theme that I don't have in the core site (i.e. `images`, `static`, `layouts`). Note these are not in the `exampleSite` folder, these are in the root of the theme. Without these files, at first look, my website works just fine with your theme but I can't tell for sure from a few tests I have done. 

This is probably a generic theme question because I have just realized that I have a similar problem with the theme I am using right now in "production" ([light-hugo](https://themes.gohugo.io/light-hugo/)) where there are a number of folders that ship with the theme (e.g. `scss`, `layouts`, `static` etc) that I did not bother to copy into the core / root site. 

Thanks!

**Reply by onweru** [reply]
Those folders that are only on the theme, they're needed for the theme to work. They're indeed overridable from the website, but you've got to know certain Hugo topics to do that. You can cherry-pick from this [quick overview/playlist](https://www.youtube.com/watch?v=qtIqKaDlqXo&list=PLLAZ4kZ9dFpOnyRlyS-liKL5ReHDcj4G3). That guide is still relevant for the most part.

**Reply by mreferre** [reply]
Thanks. I also found [this page](https://gohugo.io/getting-started/directory-structure/) useful to put those folders in context.

**Comment by vitalijalbu** [comment]
Hi guys, I don't see the search input... how come?

**Reply by mreferre** [reply]
My suggestion would be to work backwards from the example site that ships with the theme, make that your main blog and just replace the content folder with your own content folder. That worked for me (https://it20.info). Trying to keep my structure and retrofit it with this theme led me to weird behaviors.
